Mw  newsletter signup

Wmsmckay opened the pull request:
This merge adds a newsletter signup to the main page. 
To test the form:

- [x]  Make sure the email field only accepts valid emails
- [x]  Make sure that the console logs out a request with the info you put into the form

Wmsmckay committed:
added working newsletter form

Wmsmckay committed:
formatting for files

Wmsmckay committed:
removed unneeded css

JmmonJeremy commented:
So, the form only accepts a valid email so it passed that test. However, when I fill out the request with my name and email and click submit nothing appears to happen on the page and all the information is still in the boxes. It seems like the entry areas should at least clear to give an indication that it was sent. I am not sure if it is passing the second test. Is it supposed to indicate subscribed somewhere in that data that is sent? I would think the bodyUsed: should not say false, but I don't really know. Isn't that where it should indicate a request for a subscription? I am waiting to hear back before merging. The form itself looks great. Just not sure if it functions as it should when submit is clicked.

Wmsmckay committed:
added notification when user signs up

Wmsmckay commented:
I added an alert when the user signs up that just says "thank you!". I also converted the response to json and had it logged out to the console so you can see the data you "submitted"

I also added a "reset" to the form after a successful submission.

Wmsmckay committed:
added form reset after submission

Wmsmckay committed:
added formatting

JmmonJeremy commented:
I check and the form only accepts valid emails. It gives an alert thanking the user for subscribing so they know it worked and the console shows the user name and email in the body of the object, so the information is being transferred as it should be. Great work! I am merging this branch.
